src/index.ts

```diff
@@ -0,0 +1,189 @@
+ * `settings`   contains the org settings for the current session.
+ */
+type ReadyCallback = (data: { sessionUrl: string, settings: Readonly<object> }) => void;
+
+declare global {
+  interface Window {
+    _fs_cookie_domain?: string;
+    _fs_debug?: boolean;
+    _fs_dev_mode?: boolean;
+    _fs_host?: string;
+    _fs_initialized?: boolean;
+    _fs_is_outer_script?: boolean;
+    _fs_namespace?: string;
+    _fs_org?: string;
+    _fs_run_in_iframe?: boolean;
+    _fs_script?: string;
+  }
+}
+
+const getFullStory = (): FSApi | undefined => {
+  if (window._fs_namespace) {
+    return window[window._fs_namespace];
+  }
+  return undefined;
+};
+
+const ensureSnippetLoaded = (): FSApi => {
+  const fs = getFullStory();
+  if (!fs) {
+    throw Error(
+      'FullStory is not loaded, please ensure the init function is invoked before calling FullStory API functions'
+    );
+  }
+
+  return fs;
+};
+
+const _init = (inputOptions: SnippetOptions, readyCallback?: ReadyCallback) => {
+  // Make a copy so we can modify `options` if desired.
+  const options = { ...inputOptions };
+  if (getFullStory()) {
+    console.warn('The FullStory snippet has already been defined elsewhere (likely in the <head> element)');
+    return;
+  }
+
+  // see README for details on the recordCrossDomainIFrames option
+  if (options.recordCrossDomainIFrames) {
+    window._fs_run_in_iframe = true;
+  }
+
+  // record the contents of this iFrame when embedded in a parent site
+  if (options.recordOnlyThisIFrame) {
+    window._fs_is_outer_script = true;
+  }
+
+  // Set cookie domain if it was specified.
+  if (options.cookieDomain) {
+    window._fs_cookie_domain = options.cookieDomain;
+  }
+
+  if (options.debug === true) {
+    if (!options.script) {
+      options.script = 'edge.fullstory.com/s/fs-debug.js';
+    } else {
+      console.warn('Ignoring `debug = true` because `script` is set');
+    }
+  }
+
+  initFS(options);
+
+  const fs = getFullStory();
+
+  if (!fs) {
+    console.warn('Trouble with fs');
+    return;
+  }
+
+  if (readyCallback) {
+    fs('observe', { type: 'start', callback: readyCallback });
+  }
+
+  if (options.devMode === true) {
+    const message = 'FullStory was initialized in devMode and will stop recording';
+    fs('trackEvent', {
+      name: 'FullStory Dev Mode',
+      properties: {
+        message_str: message,
+      }
+    });
+    fs('shutdown');
+    window._fs_dev_mode = true;
+    console.warn(message);
+  }
+};
+
+const initOnce = (message) => (inputOptions: SnippetOptions, readyCallback?: ReadyCallback) => {
+  if (window._fs_initialized) {
+    if (message) console.warn(message);
+    return;
+  }
+  _init(inputOptions, readyCallback);
+  window._fs_initialized = true;
+};
+
+export const init = initOnce('FullStory init has already been called once, additional invocations are ignored');
+
+// normalize undefined into boolean
+export const isInitialized = () => !!window._fs_initialized;
```

Comment:
should this be an API function?

Reply:
I think I'd prefer the `FullStory` object exported from the match the Browser object as much as possible. I don't have a _strong_ feeling against this, but I might prefer it.

Additionally, we could add a `FS('isInitialized')` that returns `true` when `FS` is initialized 🤔 

Actually, we wouldn't be able to respond to this until fs.js comes back, so it wouldn't be a 1:1 match of this.

Do you know if people use this function?